File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] edge) {
           int answer = 0;
        List<List<Integer>> graph = new ArrayList<>();
        
        for(int i = 0; i < n; i++){
            graph.add(new ArrayList<>());
        }
        
        for(int[] item : edge){
            int u = item[0]; int v = item[1];
            graph.get(u-1).add(v-1);
            graph.get(v-1).add(u-1);
        }
        
        boolean[] visited = new boolean[n];
        Arrays.fill(visited, false);
        
        Queue<Integer> queue = new LinkedList<>();
        queue.add(0);
        visited[0] = true;
        
        while(!queue.isEmpty()){
            int queueSize = queue.size();
            answer = 0;
            for(int i = 0; i < queueSize; i++){
                int u = queue.poll();    
                answer++;
                
                for(int v : graph.get(u)){
                    if(visited[v]) continue;
                    visited[v] = true;
                    queue.add(v);
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.py
from collections import deque

def solution(n, edges):
    answer = 0
    graph = [[] for _ in range(n+1)]
    
    for edge in edges:
        u,v = edge
        graph[u].append(v)
        graph[v].append(u)
    
    visited = set()
    visited.add(1)
    queue = deque()
    queue.append(1)
    
    while queue:
        answer = 0
        for _ in range(len(queue)):
            curNode = queue.popleft()
            answer += 1
            
            for nearNode in graph[curNode]:
                if nearNode in visited:
                    continue
                queue.append(nearNode)
                visited.add(nearNode)
    
    return answer